sympy/calculus/util.py

```diff
@@ -1171,6 +1171,18 @@ def intersection(self, other):
                 return self
 
     def union(self, other):
+        """
+        Returns the union of 'self' and 'other'.
+        Here 'other' can be an instance of FiniteSet or AccumulationBounds.
```

Comment:
```python
In [6]: AccumBounds(1, 3).union(FiniteSet(2, 4))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-47b7c3643d5a> in <module>()
----> 1 AccumBounds(1, 3).union(FiniteSet(2, 4))

/Users/aktech/workspace/sympy/sympy/calculus/util.py in union(self, other)
   1176         # can be made better
   1177         if not isinstance(other, AccumBounds):
-> 1178             raise TypeError("Input must be AccumulationBounds or FiniteSet object")
   1179
   1180         if self.min <= other.min and self.max >= other.min:

TypeError: Input must be AccumulationBounds or FiniteSet object
```

It seems the exception message thrown is wrong. It raises `TypeError` if `other` is not an instance of `AccumBounds` as per [this](https://github.com/sympy/sympy/pull/11575/files#diff-a5d2575656fef33ceed5fc4cd005e3c0L1177) line. That means it doesn't accepts `FiniteSet`.